spaceship/lib/spaceship/portal/portal_client.rb

```diff
@@ -465,6 +465,22 @@ def provisioning_profiles(mac: false)
       end
     end
 
+    ##
+    # this endpoint is used by Xcode to fetch provisioning profiles.
+    # The response is an xml plist but has the added benefit of containing the appId of each provisioning profile.
+    def provisioning_profiles_via_xcode_api(mac: false)
```

Comment:
I had `SPACESHIP_AVOID_XCODE_API` in the client initially but decided against it.
My thinking was this: The client is just a library of valid api requests and their parsed responses. Each method makes 1 request and shouldn't need to know much about the state of the application.
As such, it should be up to the caller (i.e. `ProvisioningProfile` class) to contain the logic of deciding which request to make.

Does that seem reasonable?